src/Core/Framework/MessageQueue/ScheduledTask/README.md

```diff
@@ -0,0 +1,178 @@
+# Scheduled Task System
+
+The Scheduled Task system provides a robust framework for executing recurring background tasks in Shopware. It supports reliable scheduling, execution tracking, error handling, and automatic recovery mechanisms.
+
+## Architecture Overview
+
+The system consists of several key components:
+
+### Core Components
+
+- **ScheduledTask** (`ScheduledTask.php`) - Abstract base class for all scheduled tasks
+- **ScheduledTaskEntity** (`ScheduledTaskEntity.php`) - Entity representing task persistence in database
+- **ScheduledTaskDefinition** (`ScheduledTaskDefinition.php`) - Data definition with status constants
+- **ScheduledTaskHandler** (`ScheduledTaskHandler.php`) - Abstract handler for task execution
+- **TaskScheduler** (`Scheduler/TaskScheduler.php`) - Manages task queueing and scheduling logic
+- **TaskRunner** (`Scheduler/TaskRunner.php`) - Handles direct task execution
+- **TaskRegistry** (`Registry/TaskRegistry.php`) - Manages task registration and lifecycle
+
+### Support Components
+
+- **RegisterScheduledTaskMessage** (`MessageQueue/RegisterScheduledTaskMessage.php`) - Message for task registration
+- **RegisterScheduledTaskHandler** (`MessageQueue/RegisterScheduledTaskHandler.php`) - Handles task registration
+- **ScheduleProvider** (`SymfonyBridge/ScheduleProvider.php`) - Symfony Scheduler integration (experimental)
+
+## Task States and Lifecycle
+
+### Available States
+
+1. **`scheduled`** - Task is ready to be queued when its execution time arrives
+2. **`queued`** - Task has been dispatched to the message queue
+3. **`running`** - Task is currently being executed by a worker
+4. **`failed`** - Task execution failed and won't be rescheduled (unless `shouldRescheduleOnFailure()` returns true)
+5. **`skipped`** - Task was skipped due to configuration or conditions
+6. **`inactive`** - Task is disabled and won't be executed
+
+## State Transition Details
+
+### Initial Registration
+- **Service**: `TaskRegistry::registerTasks()`
+- **Trigger**: Application startup or task registration
+- **New State**: `scheduled` if `shouldRun()` returns true, otherwise `skipped`
+
+### Scheduled → Queued
+- **Service**: `TaskScheduler::queueScheduledTasks()`
+- **Trigger**: Task's `nextExecutionTime` has passed
+- **Conditions**: Task status is `scheduled` or `skipped`, and `shouldRun()` returns true
+- **Action**: Updates status to `queued`, dispatches task to message bus
+
+### Queued → Running
+- **Service**: `ScheduledTaskHandler::__invoke()`
+- **Trigger**: Message queue worker picks up the task
+- **Action**: Updates status to `running` before execution begins
+
+### Running → Scheduled (Success)
+- **Service**: `ScheduledTaskHandler::rescheduleTask()`
+- **Trigger**: Task execution completes successfully
+- **Action**: Updates status to `scheduled`, sets `lastExecutionTime`, calculates new `nextExecutionTime`
```

Comment:
don't you just deprecate the nextExcecutionTime thing?

Reply:
I deprecated that the task scheduler has a method to expose the next execution time of any task

Reply:
ah :bulb: I somehow thought the whole thing "next execution time" was not necessary anymore :grimacing: